crates/net/network/src/config.rs

```diff
@@ -74,6 +75,8 @@ pub struct NetworkConfig<C> {
     pub extra_protocols: RlpxSubProtocols,
     /// Whether to disable transaction gossip
     pub tx_gossip_disabled: bool,
+    /// How to instantiate transactions manager.
+    pub transactions_manager_config: TransactionsManagerConfig,
```

Comment:
let's move this from the `NetworkConfig` to the `NetworkConfigBuilder` instead, realised it makes more sense there. then the tx manager config can be added the same way that the `peer_config` is added
```rust
    /// Set a custom peer config for how peers are handled
    pub fn peer_config(mut self, config: PeersConfig) -> Self {
        self.peers_config = Some(config);
        self
    }
```

Reply:
have a question on what's best way to do it, sry this pr seems stand quite a while, i'm not very familiar with the best practices and design philosophy here.

Reply:
try putting the config in here

https://github.com/paradigmxyz/reth/blob/4a8a68b8b9ea19713d6f40ab3e77894568c993e3/crates/config/src/config.rs#L10-L24

no need to apologise. as long as it becomes good and you have fun, it doesn't matter if it takes long.

Reply:
ok, nvm, I see that the peers config is also put into the `NetworkConfig`, and yes you are quite right that we can't destruct the `NetworkConfig` because `NetworkManager` needs it in whole.

Reply:
emm, so should i put the transactionsmanagerconfig here, or is it fine to leave it in networkconfig?

Reply:
it's fine to leave it